### PR TITLE
[Core] Refactor sigterm to ray.init

### LINKED ISSUE #17745: Don't set SIGTERM handler on import
### What is the problem?

When importing Ray, any previous SIGTERM handlers are disabled due to Ray overwriting the SIGTERM handler outside of any function here: https://github.com/ray-project/ray/blob/7a0b8982f3195182f17aca54232165441689c6f5/python/ray/worker.py#L1036

A simple import thus can have large unintended side effects. For example, in our repository this caused submitit (the Slurm management system we use) to no longer properly handle job preempting, even though no functionality of Ray was actually used in those runs. Due to submitit design reasons the Ray import actually happens after submitit sets the signal handler.

Imho an import shouldn't have such side effects.

*Ray version and other system information (Python version, TensorFlow version, OS):* Ray 1.4.0, Python 3.8.10, Ubuntu 20.04

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

```
import signal
import time
signal.signal(signal.SIGTERM, signal.SIG_IGN)

from ray import tune

time.sleep(100)
# SIGTERM signal should be ignored now, but isn't due to side effects from importing ray.
# Check this via `kill -SIGTERM [pid]`
```

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by jonathan-anyscale**
## Why are these changes needed?
Resolving issue when importing ray shouldn't affect signal handler, only when we run driver process in main thread

## Related issue number

Closes #17745

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — COMMENTED** [review]
Have you discussed the solution with Jiajun btw?

**Comment by jonathan-anyscale** [comment]
> Have you discussed the solution with Jiajun btw?

Yes, but just a high level by moving the code inside the driver, but seems like there's error in other test that didn't call `ray.init` so I'll check that today

**Comment by rkooo567** [comment]
1. can we add a test (there must be a way to access sigterm handler? or you can set the sigterm handler -> import ray -> make sure the original sigterm handler is triggerd) 
2. We should update the doc (and specifies this behavior)

**Comment by jonathan-anyscale** [comment]
> 1. can we add a test (there must be a way to access sigterm handler? or you can set the sigterm handler -> import ray -> make sure the original sigterm handler is triggerd)
> 2. We should update the doc (and specifies this behavior)

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
`test_tls_auth` failure seems related

**Comment by jonathan-anyscale** [comment]
Resolved the `test_tls_auth` but `test_ray_init:test_sigterm_handler` failed on window build. Seems like window doesn't handle SIGTERM well as described here
https://stackoverflow.com/questions/35772001/how-to-handle-a-signal-sigint-on-a-windows-os-machine
Thus, will skip the test when it run on window

**Comment by rkooo567** [comment]
Do you know why it passed before this pr?

**Comment by jonathan-anyscale** [comment]
> Do you know why it passed before this pr?

It failed before as well to all commits after I added the unit test.
https://buildkite.com/ray-project/oss-ci-build-pr/builds/36093#018a8f91-5347-4e2e-a88a-e70fa9137746/3736-3919

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Lmk when build finishes!

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -1745,20 +1758,6 @@ def shutdown(_exiting_interpreter: bool = False):
 
 atexit.register(shutdown, True)
 
-
-# TODO(edoakes): this should only be set in the driver.
-def sigterm_handler(signum, frame):
-    sys.exit(signum)
```
Comment (jjyao) [inline python/ray/_private/worker.py:1751]: Actually I'm wondering why do we need this custom sigterm handler? Isn't the default behavior just exiting the process. @rkooo567 do you know why?

Comment (jonathan-anyscale) [reply python/ray/_private/worker.py:1751]: Edward told me that "presumably the registered atexit handlers don’t run without it"

Comment (rkooo567) [reply python/ray/_private/worker.py:1751]: I don't really know (this is the legacy code). I think it is safe to keep it for now, but we can try experimentation.

**Comment by rkooo567** [comment]
@jonathan-anyscale did it pass the windows build?

**Pull request merged**